src/participant/participant.ts

```diff
@@ -1084,6 +1144,34 @@ export default class ParticipantController {
       });
     }
 
+    if (!collectionName) {
+      collectionName = await this._getOrAskForCollectionName({
+        command: '/schema',
+        context,
+        databaseName,
+        stream,
+      });
+
+      // If the collection name could not get automatically selected,
+      // then the user has been prompted for it.
+      if (!collectionName) {
+        return namespaceRequestChatResult({
+          databaseName,
+          collectionName: undefined,
+          history: context.history,
+        });
+      }
+
+      // Save the collection name in the metadata.
+      const chatId = ChatMetadataStore.getChatIdFromHistoryOrNewChatId(
+        context.history
+      );
+      this._chatMetadataStore.setChatMetadata(chatId, {
```

Comment:
Does this make sense to do? If I understand right this means this will cache it but I haven't tested it yet, will soon.